Branch-SDK-TestBed/src/androidTest/java/io/branch/branchandroidtestbed/BUOTestRoutines.java

```diff
@@ -178,7 +180,8 @@ protected JSONObject doInBackground(String... strings) {
                         respObject = new JSONObject(rd.readLine());
                     }
                 }
-            } catch (Exception ignore) {
+            } catch (Exception e) {
+                BranchLogger.d(Objects.requireNonNull(e.getMessage()));
```

Comment:
Since the function is written in kotlin and uses a non-null parameter for the message, we'll need to use the `Objects.requireNonNull` assertion when passing in objects not explicitly marked `@Non-Null` correct?

Reply:
Yes, I want to discuss it.

we can take a call as let null value pass in the function and then check there or check in the first place.
We can make the message value nullable like below, this will help us in code repetitiveness also.

```
fun d(message: String?) {
        if (loggingEnabled && message?.isNotEmpty() == true) {
            Log.d(TAG, message)
        }
    }
```

Reply:
This change only requires .d() because usually, we call it from the exception block. So will make this change.

Reply:
Yup I think let's change to `String?` and check `message?.isNotEmpty()`